tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/HTTPCachingTestCase.java

```diff
@@ -61,4 +61,45 @@ public void testPassthroughServiceByBasePath() throws IOException, InterruptedEx
         assertEquals(response.getHeaders().get(proxyHitCount), "3");
         assertEquals(response.getData(), payload);
     }
+
+    @Test(description = "Test no-cache cache control")
+    public void testNoCacheCacheControl() throws IOException {
+        HttpResponse response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9244, "nocache"));
+        assertEquals(response.getData(), "{\"message\":\"1st response\"}");
+        assertEquals(response.getHeaders().get(serviceHitCount), "1");
+
+        response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9244, "nocache"));
+        assertEquals(response.getData(), "{\"message\":\"2nd response\"}");
+        assertEquals(response.getHeaders().get(serviceHitCount), "2");
+
+        response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9244, "nocache"));
+        assertEquals(response.getData(), "{\"message\":\"2nd response\"}");
+        assertEquals(response.getHeaders().get(serviceHitCount), "3");
+    }
+
+    @Test(description = "Test max-age cache control")
+    public void testMaxAgeCacheControl() throws IOException, InterruptedException {
+        HttpResponse response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9245, "maxAge"));
+        assertEquals(response.getData(), "public,max-age=5");
+
+        response = HttpClientRequest.doGet(serverInstance.getServiceURLHttp(9245, "maxAge"));
+        assertEquals(response.getData(), "public,max-age=5");
+
+        Thread.sleep(5000);
```

Comment:
Guys, for tests with thread sleep, shall we follow the convention mentioned bellow,
https://martinfowler.com/articles/nonDeterminism.html

Reply:
In here, Thread.sleep is not added to wait for a response. We are sending the request after 5 seconds. So that the max-age would expire, and we need to test the response after the cache expiration.